### DISCUSSION TITLE
Web Only Templates

### DISCUSSION THREAD

**Discussion by tyrauber** [expo/router#129] [General]
Any thoughts on platform specific layouts and pages?

```
pages/
  _layout.tsx
  _layout.web.tsx
  index.tsx
  index.web.tsx
```

It would be helpful to customize the UI per platform while still maintaining a single cross-platform architecture.  For example, hiding tabs, or showing a nav bar depending upon web or mobile.

**Comment by cseelus** [comment]
This would be a must have, especially for screens.

To get a different layout for web and native, inside my `_layout` file, I'm using something like this, with `WebLayout` being a component that defines a [custom layout for web, wrapped in `<Layout>`](https://expo.github.io/router/docs/features/layouts#custom):

```javascript
      {Platform.OS === 'web' ? (
        <WebLayout />
      ) : (
        <Stack
          screenOptions={{
            contentStyle: {
              backgroundColor: mainStyles.colors.background,
            },
            …
          }}
        />
      )}
```

**Comment by EvanBacon** [comment]
I think using `Platform.OS` inside the `_layout.js` makes the most sense. Could explore adding platform extensions in the future, but right now it just increases the complexity of the naming convention.

**Reply by RWOverdijk** [reply]
By "platform extensions" do you mean the `*.web.js` file naming? That would be neat, but I can see a simple Platform.select work as well.

**Reply by erisvaldojunior** [reply]
I agree Platform extensions (.web.tsx files) would be super helpful.

**Comment by cseelus** [comment]
Using some code akin to the following snippet for now

```javascript
// Layout component for authenticated users.
// It uses different layouts for web and native.
export default function Layout() {
  return (
    Platform.OS === 'web' ? <LayoutWeb /> : <LayoutNative />
  )
}
```
